LambdaEngine/Source/Engine/EngineConfig.cpp

```diff
@@ -32,8 +34,98 @@ namespace LambdaEngine
         return true;
     }
 
+    bool EngineConfig::WriteToFile()
+    {
+        const char* pEngineConfigPath = "../engine_config.json";
+
+        FILE* pFile = fopen(pEngineConfigPath, "w");
+        if (!pFile)
+        {
+            LOG_WARNING("Engine config could not be opened: %s", pEngineConfigPath);
+            return false;
+        }
+
+        char writeBuffer[2048];
+        FileWriteStream outputStream(pFile, writeBuffer, sizeof(writeBuffer));
+
+        PrettyWriter<FileWriteStream> writer(outputStream);
+        s_ConfigDocument.Accept(writer);
+
+        fclose(pFile);
+
+        return true;
+    }
+
     bool EngineConfig::GetBoolProperty(const String& propertyName)
     {
         return s_ConfigDocument[propertyName.c_str()].GetBool();
     }
+    float EngineConfig::GetFloatProperty(const String& propertyName)
+    {
+        return s_ConfigDocument[propertyName.c_str()].GetFloat();
+    }
+    int EngineConfig::GetIntProperty(const String& propertyName)
+    {
+        return s_ConfigDocument[propertyName.c_str()].GetInt();
+    }
+
+    String EngineConfig::GetStringProperty(const String& propertyName)
+    {
+        return s_ConfigDocument[propertyName.c_str()].GetString();
+    }
+
+    TArray<int> EngineConfig::GetArrayProperty(const String& propertyName)
+    {
+        const Value& arr = s_ConfigDocument[propertyName.c_str()];
+        TArray<int> tArr;
+        for (auto& itr : arr.GetArray())
+            tArr.PushBack(itr.GetInt());
+
+        return tArr;
+    }
+
+    bool EngineConfig::SetBoolProperty(const String& propertyName, const bool& value)
+    {
+        s_ConfigDocument[propertyName.c_str()].SetBool(value);
+
+        return true;
+    }
+
+    bool EngineConfig::SetFloatProperty(const String& propertyName, const float& value)
+    {
+        s_ConfigDocument[propertyName.c_str()].SetFloat(value);
+
+        return true;
+    }
+
+    bool EngineConfig::SetIntProperty(const String& propertyName, const int& value)
+    {
+        s_ConfigDocument[propertyName.c_str()].SetFloat(value);
+
+        return true;
+    }
+
+    bool EngineConfig::SetStringProperty(const String& propertyName, const String& string)
+    {
+        s_ConfigDocument[propertyName.c_str()].SetString(string.c_str(), static_cast<SizeType>(strlen(string.c_str())), s_ConfigDocument.GetAllocator());
+
+        return true;
+    }
+
+    bool EngineConfig::SetArrayProperty(const String& propertyName, const TArray<int>& arr)
+    {
+        Document::AllocatorType& allocator = s_ConfigDocument.GetAllocator();
+        s_ConfigDocument[propertyName.c_str()].SetArray().Clear();
```

Comment:
1. Pre-allocate the array before pushing values into it. That way, the array doesn't have to repeatedly resize and reallocate memory.
2. Is .Clear() needed?
3. 'Cache' a reference to the array instead of searching for it using the property name each time you want to access it.

Suggestions 1 & 2:
```suggestion
        auto& newArr = s_ConfigDocument[propertyName.c_str()].SetArray();
        newArr.Reserve(arr.GetSize());
```